Convert all timestamps to fix the comparison

TeXhackse opened the pull request:
I was thinking about opening an issue but guess it's easier to explain that way … 

If you want an issue in addition, I can open one s well.  This may be related to #401 depends a bit on the specific setup.

Actually the test for the timestamp is failing in some setups, e.g. using lualatex. The timestamp to compare with is a string and the one read out of the config file will contain letters. As `\ifx` is comparing both the test will return false, though the timestamps would match.  One option to fix this would be replacing the test via `\ifx` by something different, e.g. `\str_if_eq` but to match the rest of the packages structure I decided to do it the `\detokenize` way. 


Offtopic, but as I already write here:
Thanks for all the work! I love this! And I do even more now that shell escape is no longer required. So I will no longer have to create weird workarounds for limited setups.

gpoore commented:
Thanks for finding this! Can you give me a specific example that will cause the issue? lualatex is working for me, so there's probably some sort of dependence on TeX distribution, LuaTeX version, operating system, etc. Depending on where the issue is originating, I may need to add additional `\detokenize` elsewhere.

TeXhackse commented:
Yes it depends on the OS. I had the issue within docker containers. Still it should just not be possible that this works out. That's why I just converted al of them no matter what they are before that to strings.

I can send version numbers later this day, just don't think this really helps. As I don't know if there are other systems where the same might happen. That's why I checked all variables realizing it only happens with the timestamps and converted those everywhere.

The saver variant probably would be to just do all those comparisons as strings and not via `\ifx`.

muzimuzhi commented on latex/minted/minted.dtx:
```diff
@@ -1805,6 +1805,10 @@
 %    \end{macrocode}
+% To unify the structure of this macro to be able to compare it using |\ifx| it has to be converted to a string.
+%    \begin{macrocode}
+\xdef\minted@timestamp{\expandafter\detokenize\expandafter{\minted@timestamp}}
```
The `\expandafter` before `\detokenize` is redundant. There are 3 such cases added by this commit.

The form
```tex
\xdef\minted@timestamp{\detokenize\expandafter{\minted@timestamp}}
```
is already used in 15cd9144ac507235c1ef4aad7ea86985a9e8a7f8.

TeXhackse replied:
you are right - thats fine then.

TeXhackse commented:
So I had the issue in the IoT docker container image. It's a vanilla TeX Live on Debian. 

Version numbers would be 
```
This is LuaHBTeX, Version 1.18.0 (TeX Live 2024) 
LaTeX2e <2024-06-01> patch level 2
L3 programming layer <2024-09-10>
```
With the 3.0.0 release of minted. The timestamp there had `z` as category 12.

TeXhackse commented:
I will close this as it's already included via the other commit then. Thanks!

gpoore commented:
@TeXhackse Thanks for the additional details about the docker container and operating system! The commit I created to address this covers all the cases you found, and also tries to eliminate all potential for similar issues in the future involving config values other than timestamps (such as version numbers).
